### ISSUE TITLE
bug(web): 'wrong private key type' when initializing authenticated turbo client

### ISSUE THREAD

**Issue by gregogun** [ardriveapp/turbo-sdk#53]
**Describe the bug**
When initializing the turbo client (currently generating a jwk on-the-fly), I get the following error: 'wrong private key type'

**To Reproduce**

1. Import turbo client `import { TurboFactory } from "@ardrive/turbo-sdk"`
2. Create JWK instance `const jwk = await arweave.wallets.generate()`
3. Initialize turbo `const turbo = TurboFactory.authenticated({
privateKey: jwk
})`

**Expected behavior**
I expect to not receive an error when I'm providing a `JWKInterface` as required

**Screenshots**
Error in console with full stack trace:
<img width="543" alt="Screenshot 2023-11-29 at 10 49 24" src="https://github.com/ardriveapp/turbo-sdk/assets/65421744/aced2f00-eccc-46d0-8409-8d0dabf6c1e6">

**Environment:**

- Environment: browser
- SDK Version: 1.1.0
- Browser: Arc (chromium), Chrome
- Node: 20.3.1
- Framework: Next.js 13.1.6
- Arweave SDK Version: 1.13.7

**Additional context**
Full context of the implementation where I'm getting this error:

```
import {
  TurboFactory,
  developmentTurboConfiguration,
} from "@ardrive/turbo-sdk";
import { arweave } from "./arweave";

export const getTurboBalance = async () => {
  const jwk = await arweave.wallets.generate();
  const turbo = TurboFactory.authenticated({
    privateKey: jwk,
    ...developmentTurboConfiguration,
  });

  const balance = await turbo.getBalance();
  console.log(balance);

  return balance;
};
```

**Comment by dtfiedler** [comment]
@gregogun - can you share your `arweave` configuration. i'm having no issues creating in node environment

```typescript
  const arweave = new Arweave({});
  const jwk = await arweave.wallets.generate();
  const address = await arweave.wallets.jwkToAddress(jwk);

  TurboFactory.setLogLevel('debug');
  TurboFactory.setLogFormat('json');
  /**
   * Use the arweave key to create an authenticated turbo client
   */
  const turbo = TurboFactory.authenticated({
    privateKey: jwk,
    ...developmentTurboConfiguration,
  });
  const balance = await turbo.getBalance();
```
- it may be something web specific when generating a wallet with arweave

**Comment by gregogun** [comment]
thanks for the suggestion @dtfiedler - I tried this and got an error when trying to use the web export `Cannot find module '@ardrive/turbo-sdk/web' or its corresponding type declarations.` 

I have a temporary workaround in my tsconfig. I'll share the whole config in case that helps: 

```
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "baseUrl": ".",
    "paths": {
      "@/*": ["./*"],
      "arconnect": ["./types/arconnect.d.ts"],
      "@ardrive/turbo-sdk/web": [
        "./node_modules/@ardrive/turbo-sdk/lib/types/web"
      ]
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"]
}
```

**Comment by gregogun** [comment]
also, I'm wondering if there is a way to authenticate the client using either arconnect or arweave.app? happy to create a new issue for that though if necessary!

**Comment by dtfiedler** [comment]
> thanks for the suggestion @dtfiedler - I tried this and got an error when trying to use the web export `Cannot find module '@ardrive/turbo-sdk/web' or its corresponding type declarations.`
> 
> I have a temporary workaround in my tsconfig. I'll share the whole config in case that helps:
> 
> ```
> {
>   "compilerOptions": {
>     "target": "es5",
>     "lib": ["dom", "dom.iterable", "esnext"],
>     "allowJs": true,
>     "skipLibCheck": true,
>     "strict": true,
>     "forceConsistentCasingInFileNames": true,
>     "noEmit": true,
>     "esModuleInterop": true,
>     "module": "esnext",
>     "moduleResolution": "node",
>     "resolveJsonModule": true,
>     "isolatedModules": true,
>     "jsx": "preserve",
>     "incremental": true,
>     "baseUrl": ".",
>     "paths": {
>       "@/*": ["./*"],
>       "arconnect": ["./types/arconnect.d.ts"],
>       "@ardrive/turbo-sdk/web": [
>         "./node_modules/@ardrive/turbo-sdk/lib/types/web"
>       ]
>     }
>   },
>   "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
>   "exclude": ["node_modules"]
> }
> ```

this is a reasonable workaround - we'll look into the best options to support your tsconfig and at the very least add this workaround to the README.

as for arconnect/arweave.app support - we are actively looking into implementations but feel free to create the issue as a feature and we can update it as we progress.

**Comment by gregogun** [comment]
@dtfiedler - so turns out even with the above workaround I'm getting the following error when trying to reference the web sdk

<img width="963" alt="Screenshot 2023-12-05 at 23 19 54" src="https://github.com/ardriveapp/turbo-sdk/assets/65421744/52647511-fdea-4bba-b5ca-f6626f021b2f">

**Comment by dtfiedler** [comment]
Related: https://github.com/ardriveapp/turbo-sdk/issues/56

**Comment by dtfiedler** [comment]
@gregogun - we've pushed some updates for our package exports. they are available in [1.1.1.alpha-2 if you are interested in testing](https://github.com/ardriveapp/turbo-sdk/releases/tag/v1.1.1-alpha.2). you should be able to remove your alias and use web exports as described above.

**Comment by dtfiedler** [comment]
> also, I'm wondering if there is a way to authenticate the client using either arconnect or arweave.app? happy to create a new issue for that though if necessary!

@gregogun - arconnect and arweave signer support is now available in v1.3.0. will close this issue.

**Comment by gregogun** [comment]
@dtfiedler - awesome, thanks for the update! Is there an example or working implementation of this anywhere?
